collect/src/main/kotlin/com/certora/collect/HashTreapMap.kt

```diff
@@ -31,6 +31,7 @@ internal class HashTreapMap<@Treapable K, V>(
         this as? HashTreapMap<K, V>
         ?: (this as? PersistentMap.Builder<K, V>)?.build() as? HashTreapMap<K, V>
 
+    override fun singleOrNull() = MapEntry(key, value).takeIf { next == null && left == null && right == null }
```

Comment:
you might wonder why I don't object to this, where I am objecting to the `also` nonsense. I confess I don't have a great rule of thumb, except this looks more idiomatic, and the `takeIf` is clearly indicating what's happening, whereas using `also` to exceptionally abort is pretty surprising